### DISCUSSION TITLE
Is possible to change outputDevice audio source or instrument?

### DISCUSSION THREAD

**Discussion by tp1415926535** [melanchall/drywetmidi#274] [Q&A] [answered]
The program works fine, I'm now using the "Microsoft GS Wavetable Synth" as outputDevice, it seems like the Piano Tone.
And I wanna know how to change the source to play another sytle, maybe like using other ".sf3"/".sf2" file or VST dll ?

**Comment by melanchall** [comment, marked as answer]
DryWetMIDI doesn't provide working with audio and SoundFont files. There is no synthesizer functionality within the library. You can implement your own `IOutputDevice` with using of other software. See this comment please: https://github.com/melanchall/drywetmidi/discussions/258#discussioncomment-6040507

**Reply by tp1415926535** [reply]
Thanks for reply. After searching for information on this, it looks like it's going to be more complicated to implement than I thought.

**Comment by tp1415926535** [comment]
> DryWetMIDI doesn't provide working with audio and SoundFont files. There is no synthesizer functionality within the library. You can implement your own `IOutputDevice` with using of other software. See this comment please: [#258 (reply in thread)](https://github.com/melanchall/drywetmidi/discussions/258#discussioncomment-6040507)

But as far as I know, Midi files can set different instruments for tracks. Is there a way to achieve this?

**Reply by melanchall** [reply]
Yes, just add `ProgramChangeEvent` where you need in a track chunk.
